xrpl/transaction/main.py

```diff
@@ -104,194 +112,11 @@ def safe_sign_and_autofill_transaction(
-    """
-    transaction_json = transaction.to_xrpl()
-    transaction_json["SigningPubKey"] = wallet.public_key
-
-    _validate_account_xaddress(transaction_json, "Account", "SourceTag")
-    if "Destination" in transaction_json:
-        _validate_account_xaddress(transaction_json, "Destination", "DestinationTag")
-
-    # DepositPreauth
-    _convert_to_classic_address(transaction_json, "Authorize")
-    _convert_to_classic_address(transaction_json, "Unauthorize")
-    # EscrowCancel, EscrowFinish
-    _convert_to_classic_address(transaction_json, "Owner")
-    # SetRegularKey
-    _convert_to_classic_address(transaction_json, "RegularKey")
-
-    return transaction_json
-
-
-def _autofill_transaction(transaction: Transaction, client: Client) -> Transaction:
-    transaction_json = transaction.to_dict()
-    if "sequence" not in transaction_json:
-        sequence = get_next_valid_seq_number(transaction_json["account"], client)
-        transaction_json["sequence"] = sequence
-    if "fee" not in transaction_json:
-        transaction_json["fee"] = _calculate_fee_per_transaction_type(
-            transaction, client
+    return asyncio.run(
+        main.safe_sign_and_autofill_transaction(
+            transaction,
+            wallet,
+            client,
+            check_fee,
         )
-    if "last_ledger_sequence" not in transaction_json:
-        ledger_sequence = get_latest_validated_ledger_sequence(client)
-        transaction_json["last_ledger_sequence"] = ledger_sequence + _LEDGER_OFFSET
-    return Transaction.from_dict(transaction_json)
-
-
-def _validate_account_xaddress(
-    json: Dict[str, Any], account_field: str, tag_field: str
-) -> None:
-    """
-    Mutates JSON-like dictionary so the X-Address in the account field is the classic
-    address, and the tag is in the tag field.
-    """
-    if is_valid_xaddress(json[account_field]):
-        account, tag, _ = xaddress_to_classic_address(json[account_field])
-        json[account_field] = account
-        if json[tag_field] and json[tag_field] != tag:
-            raise XRPLException(f"{tag_field} value does not match X-Address tag")
-        json[tag_field] = tag
-
-
-def _convert_to_classic_address(json: Dict[str, Any], field: str) -> None:
-    """
-    Mutates JSON-like dictionary to convert the given field from an X-address (if
-    applicable) to a classic address.
-    """
-    if field in json and is_valid_xaddress(json[field]):
-        json[field] = xaddress_to_classic_address(json[field])
-
-
-def transaction_json_to_binary_codec_form(dictionary: Dict[str, Any]) -> Dict[str, Any]:
-    """
-    Returns a new dictionary in which the keys have been formatted as CamelCase and
-    standardized to be serialized by the binary codec.
-
-    Args:
-        dictionary: The dictionary to be reformatted.
-
-    Returns:
-        A new dictionary object that has been reformatted.
-    """
-    return model_transaction_to_binary_codec(dictionary)
```

Comment:
This method should still be there for backwards compatibility

Reply:
i've left it there. it's defined in the asyncio package but reimported/exported in the __init__ file of this pacakge too

Reply:
I thought you said that wasn't a very good way of handling that.

Reply:
i did? that's exactly what we're doing in your other branch though. it's implemented in the model now, but we're exporting it from here too for backwards compat